sdk/Pulumi/Provider/Provider.cs

```diff
@@ -504,6 +504,15 @@ public virtual Task<CallResponse> Call(CallRequest request, CancellationToken ct
 
         public static async Task Serve(string[] args, string? version, Func<IHost, Provider> factory, System.Threading.CancellationToken cancellationToken)
         {
+            var value = System.Environment.GetEnvironmentVariable("PULUMI_ATTACH_DEBUGGER");
```

Comment:
I know this is just a continuation of the way we do this for programs, but urgh it feels grim.

Reply:
Yeah, it's kinda sad, but unfortunately I do not know a better way.